proposals/async-method-builders.md

```diff
@@ -80,52 +100,131 @@ static ValueTask<int> ExampleAsync()
 ```
 
 Just those small additions enable:
-- Anyone to write their own builder that can be applied to async methods that return `Task{T}` and `ValueTask{<T>}`
+- Anyone to write their own builder that can be applied to async methods that return `Task<T>` and `ValueTask<T>`
 - As "anyone", the runtime to ship the experimental builder support as new public builder types that can be opted into on a method-by-method basis; the existing support would be removed from the existing builders.  Methods (including some we care about in the core libraries) can then be attributed on a case-by-case basis to use the pooling support, without impacting any other unattributed methods.
 
 and with minimal surface area changes or feature work in the compiler.
 
-#### P1: AsyncMethodBuilderAttribute arguments forward to Create
+#### P1: Passing state to the builder instantiation
+
+In some scenarios, it would be useful to pass some information to the builder.  This could be static (e.g. constants configuring the size of the pool to use) or dynamic (e.g. reference to cache or singleton to use).
+
+We brainstormed a few options (documented below for the record) but end up recommending doing nothing (option #0) until we determine that supporting dynamic state would be worthwhile.
+
+##### Option 0: no extra state
+
+It is possible to approximate passing static information by wrapping builder types.
+For instance, one could create a custom builder type that hardcodes a certain configuration:
 
-The attribute would be given an additional constructor:
 ```C#
-public AsyncMethodBuilderAttribute(Type builderType, params object[] createArguments);
+public struct AsyncPooledBuilderWithSize4
+{
+    AsyncPooledBuilder Create() => AsyncPooledBuilder.Create(4);
+}
 ```
-If any such arguments are specified, the compiler would expect the builder to have a `Create` method that could bind with those arguments, e.g. if a developer used:
+
+Note that we need the emitted code to allow a different type being returned from `Create` method:
```

Comment:
I think we should call this out as its own thing.  It's useful regardless of which of these options is chosen, basically just making it possible to author a builder factory.